workers/price-reporter/src/manager/external_executor.rs

```diff
@@ -225,19 +225,48 @@ impl ExternalPriceReporterExecutor {
         quote_token: Token,
         msg_out_tx: UnboundedSender<WebsocketMessage>,
     ) -> Result<(), PriceReporterError> {
-        // TODO: Conditional price conversion
-
         let exchanges = get_supported_exchanges(&base_token, &quote_token, &self.config);
         for exchange in exchanges {
-            subscribe_to_price_stream(
-                &self.subscription_states,
-                exchange,
-                base_token.clone(),
-                quote_token.clone(),
-                msg_out_tx.clone(),
-            )
-            .await
-            .map_err(PriceReporterError::ExchangeConnection)?;
+            // If the quote is a stablecoin (and the base is not), we may invoke price
+            // conversion through the default stable quote for the exchange
+            if eligible_for_stable_quote_conversion(&base_token, &quote_token, &exchange) {
```

Comment:
Lots of repeat code here, why not 
```suggestion
            let subs = if eligible_for_stable_quote_conversion(&base_token, &quote_token, &exchange) {
                let default_stable = default_exchange_stable(&exchange);
                [(base_token, default_stable), (quote_token, default_stable)]
            } else {
                [(base_token, quote_token)]
            };
            
            for (base, quote) in subs.iter() {
                // Subscribe
            }
```